### PR TITLE
num_workers bug in gpt prompt learning dataset bug

### PR DESCRIPTION
**Pull request by arendu**
# What does this PR do ?

fixes the bug where nw0 does weird things in training and validation after the first epoch.

**Collection**: [NLP]

# Changelog 
- changes gpt dataset creation by deep copy of inputs.

# Usage

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [X] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py [hunk 1]
```diff
@@ -122,8 +123,9 @@ def load_data(self, dataset):
         """
         skipped = 0
 
+        ex_id = -1
```
Comment (vadam5) [inline nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py:126]: Can we remove all the ex_id stuff?

File: examples/nlp/language_modeling/conf/megatron_gpt_prompt_learning_config.yaml [hunk 2]
```diff
@@ -90,6 +90,14 @@ model:
     truncate_field: null
     answer_only_loss: True
     answer_field: "answer"
+  
+  - taskname: "squad"
```
Comment (vadam5) [inline examples/nlp/language_modeling/conf/megatron_gpt_prompt_learning_config.yaml:94]: I think we should switch to just having squad in the example config, instead of the other 3 tasks. But we should probably do that in a different PR.

**Review by vadam5 — COMMENTED** [review]
Thank you! Left 2 comments.

File: nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py (hunk 1 above)
Comment (arendu) [reply nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py:126]: it is nice to have the actual example id for debugging things like this. ofc, its easy to remove if you feel strongly.

**Comment by okuchaiev** [comment]
@ericharper @arendu what's up with base branch? Shouldn't this go into r1.13. Then r1.13=>main?
